CHANGES.rst

```diff
@@ -15,6 +15,14 @@ Version 3.1.0 (2019-XX-XX)
     + ``io.eager.update_dataset_from_dataframes``
 - fix ``_apply_partition_key_predicates`` ``FutureWarning``
 - serialize :class:`~kartothek.core.index.ExplicitSecondaryIndex` to parquet
+- Provide compatibility for dataset schemas created in Python2:
```

Comment:
I wouldn't promote this as a py2 compatibility feature. This might open the doors for support we explicitly don't want to give.
We have a small feature within kartothek that we ensure that columns always have `string` type and not `binary`. This PR rather fixes this for edge cases where the metadata was stored with columns of binary type (which **coincidentally** was the case in Py2)